### ISSUE TITLE
.NET 5.0 Preview 8 - Blazor Web Assembly - InputSelect Exception

### ISSUE THREAD

**Issue by boukenka** [dotnet/aspnetcore#25728]
### Description

I am getting the exception below when selecting a value in the drop down with the following code.
This only happens if the website is running from a container (LINUX).
There is no exception when the site is running on IIS Express.
The values are GUID type.

```
<InputSelect id="servertype" @bind-Value="AddRequest.STypeId">
<option disabled selected>Select server type</option>
@foreach (var sTypeResponse in sTypeResponses)
{
   <option value="@sTypeResponse.Id" selected="@(sTypeResponse.Id == AddRequest.STypeId)">
    @sTypeResponse.Name
   </option>
 }
</InputSelect>
```

I could fix the issue by adding the following in the project. After that there was no problem.

```
<PublishTrimmed>false</PublishTrimmed>
```

I consider this to be a workaround. Could you have a look at this issue please ?

_System.MissingMethodException: Default constructor not found for type System.ComponentModel.NullableConverter
   at System.RuntimeType.CreateInstanceMono(Boolean nonPublic, Boolean wrapExceptions)
   at System.RuntimeType.CreateInstanceSlow(Boolean publicOnly, Boolean wrapExceptions, Boolean skipCheckThis, Boolean fillCache)
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, Boolean wrapExceptions)
   at System.Activator.CreateInstance(Type type, Boolean nonPublic, Boolean wrapExceptions)
   at System.Activator.CreateInstance(Type type, Boolean nonPublic)
   at System.Activator.CreateInstance(Type type)
   at System.ComponentModel.ReflectTypeDescriptionProvider.CreateInstance(Type objectType, Type callingType)
   at System.ComponentModel.ReflectTypeDescriptionProvider.SearchIntrinsicTable(Hashtable table, Type callingType)
   at System.ComponentModel.ReflectTypeDescriptionProvider.ReflectedTypeData.GetConverter(Object instance)
   at System.ComponentModel.ReflectTypeDescriptionProvider.GetConverter(Type type, Object instance)
   at System.ComponentModel.TypeDescriptor.TypeDescriptionNode.DefaultTypeDescriptor.System.ComponentModel.ICustomTypeDescriptor.GetConverter()
   at System.ComponentModel.TypeDescriptor.GetConverter(Type type)
   at Microsoft.AspNetCore.Components.BindConverter.ParserDelegateCache.MakeTypeConverterConverter[Nullable`1]()
   at Microsoft.AspNetCore.Components.BindConverter.ParserDelegateCache.Get[Nullable`1]()
   at Microsoft.AspNetCore.Components.BindConverter.TryConvertTo[Nullable`1](Object obj, CultureInfo culture, Nullable`1& value)
   at Microsoft.AspNetCore.Components.Forms.InputExtensions.TryParseSelectableValueFromString[Nullable`1](InputBase`1 input, String value, Nullable`1& result, String& validationErrorMessage)
   at Microsoft.AspNetCore.Components.Forms.InputSelect`1[[System.Nullable`1[[System.Guid, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].TryParseValueFromString(String value, Nullable`1& result, String& validationErrorMessage)
   at Microsoft.AspNetCore.Components.Forms.InputBase`1[[System.Nullable`1[[System.Guid, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].set_CurrentValueAsString(String value)
   at Microsoft.AspNetCore.Components.Forms.InputSelect`1[[System.Nullable`1[[System.Guid, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].<BuildRenderTree>b__4_0(String __value)
   at Microsoft.AspNetCore.Components.EventCallbackFactoryBinderExtensions.<>c__DisplayClass24_0`1[[System.String, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].<CreateBinderCore>b__0(ChangeEventArgs e)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
--- End of stack trace from previous location ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Microsoft.AspNetCore.Components.ComponentBase.<CallStateHasChangedOnAsyncCompletion>d__22.MoveNext()
--- End of stack trace from previous location ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.<GetErrorHandledTask>d__49.MoveNext() blazor.webassembly.js:1:37220
    printErr http://localhost:8000/_framework/blazor.webassembly.js:1
    dotNetCriticalError http://localhost:8000/_framework/blazor.webassembly.js:1
    _mono_wasm_invoke_js_unmarshalled http://localhost:8000/_framework/dotnet.5.0.0-preview.8.20407.11.js:1
    mono_wasm_invoke_js_unmarshalled http://localhost:8000/_framework/dotnet.wasm:0
    do_icall http://localhost:8000/_framework/dotnet.wasm:1720439
    do_icall_wrapper http://localhost:8000/_framework/dotnet.wasm:509469
    interp_exec_method http://localhost:8000/_framework/dotnet.wasm:288284
    interp_runtime_invoke http://localhost:8000/_framework/dotnet.wasm:1289371
    mono_jit_runtime_invoke http://localhost:8000/_framework/dotnet.wasm:1194987
    do_runtime_invoke http://localhost:8000/_framework/dotnet.wasm:509286
    mono_runtime_invoke_checked http://localhost:8000/_framework/dotnet.wasm:65671
    mono_runtime_try_invoke_array http://localhost:8000/_framework/dotnet.wasm:1151366
    ves_icall_InternalInvoke http://localhost:8000/_framework/dotnet.wasm:1010156
    ves_icall_InternalInvoke_raw http://localhost:8000/_framework/dotnet.wasm:1008816
    do_icall http://localhost:8000/_framework/dotnet.wasm:1720324
    do_icall_wrapper http://localhost:8000/_framework/dotnet.wasm:509469
    interp_exec_method http://localhost:8000/_framework/dotnet.wasm:288284
    interp_runtime_invoke http://localhost:8000/_framework/dotnet.wasm:1289371
    mono_jit_runtime_invoke http://localhost:8000/_framework/dotnet.wasm:1194987
    do_runtime_invoke http://localhost:8000/_framework/dotnet.wasm:509286
    mono_runtime_try_invoke http://localhost:8000/_framework/dotnet.wasm:84530
    mono_runtime_invoke http://localhost:8000/_framework/dotnet.wasm:1152237
    mono_wasm_invoke_method http://localhost:8000/_framework/dotnet.wasm:1128384
    _mono_wasm_invoke_method http://localhost:8000/_framework/dotnet.5.0.0-preview.8.20407.11.js:1
    call_method http://localhost:8000/_framework/dotnet.5.0.0-preview.8.20407.11.js:1
    bind_static_method http://localhost:8000/_framework/dotnet.5.0.0-preview.8.20407.11.js:1
    beginInvokeDotNetFromJS http://localhost:8000/_framework/blazor.webassembly.js:1
    u http://localhost:8000/_framework/blazor.webassembly.js:1
    invokeMethodAsync http://localhost:8000/_framework/blazor.webassembly.js:1
    v http://localhost:8000/_framework/blazor.webassembly.js:1
    invokeWhenHeapUnlocked http://localhost:8000/_framework/blazor.webassembly.js:1
    v http://localhost:8000/_framework/blazor.webassembly.js:1
    dispatchEvent http://localhost:8000/_framework/blazor.webassembly.js:1
    eventDelegator http://localhost:8000/_framework/blazor.webassembly.js:1
    eventDelegator http://localhost:8000/_framework/blazor.webassembly.js:1
    onGlobalEvent http://localhost:8000/_framework/blazor.webassembly.js:1_

### Further technical details

ASP.NET Core 5.0 Preview 8
Firefox 80.0.1
Visual Studio 16.8.0 Preview 2.1
Blazor Web Assembly + PWA + HTTPS + ASP.NET Core Hosted

**Comment by pranavkm** [comment]
FYI @layomia \ @eerhardt

**Comment by eerhardt** [comment]
This was fixed in 5.0 RC1. See https://github.com/dotnet/runtime/pull/39973

**Comment by boukenka** [comment]
@pranavkm @eerhardt Thank you
